daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -1097,6 +1097,9 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
             arg = decodeExpr(create.getExpr, definition))
 
         case PLF.Update.SumCase.EXERCISE =>
+          if (versionIsOlderThan(LV.Features.noExerciseActor)) {
```

Comment:
Wouldn't it make sense to check for the minimal supported version before we even start decoding a package?

Reply:
Good point.